src/saving/SaveHelper.cs

```diff
@@ -127,7 +127,24 @@ public static void QuickLoad()
             return;
         }
 
-        LoadSave(save);
+        var info = global::Save.LoadJustInfoFromSave(save);
+        var versionDiff = VersionUtils.Compare(info.ThriveVersion, Constants.Version);
+        if (versionDiff != 0)
+        {
+            Task.Run(async () =>
+            {
+                LoadingScreen.Instance.Show($"Tried to load {(versionDiff > 0 ? "a newer" : "an older")} save.\n" +
+                    $"The current version is {Constants.Version} " +
+                    $"but you tried to load a {info.ThriveVersion} save.\n",
+                    "Please load it manually through the menu.\nWait to go back.");
+                await Task.Delay(TimeSpan.FromSeconds(5)).ConfigureAwait(true);
+                LoadingScreen.Instance.Hide();
```

Comment:
First time I'm seeing this kind of black magic stuff.
I think there should be just a normal dialog box with an ok button to close it.
Also the text doesn't need to say which versions and stuff, just saying "it is a different version, load it from the load game menu" is enough in my opinion.

For that it's maybe easier to show in the auto load from which QuickLoad is called.
I imagined that the version check code that you put here, would be there anyway, well the "get version difference with latest save" can be in this class and called by the quick load.

Reply:
If I read this correctly, you want me to move the validation code from QuickLoad to the caller(QuickLoadHandler). Is that correct?

Reply:
Yes, that should easily allow making a new dialog popup for the error message.

Reply:
I'm sorry, but do you know how to open a dialog popup from an AutoLoad class? I couln't find anything in the internet.

Reply:
I think you just add a Control child node, and then put it in there. You might need additionally need to tweak the overlay manager to make it position the quick load handler properly so that it is after the other GUI content in order to be visible (check for example the colour blind filter on how it does it).

Reply:
Maybe I'm really stupid right now, but how can I create a Child node in an AutoLoad class? I mean, it doesn't have any tscn file associated with it

Reply:
You can autoload scene files as well, for example this is an autoload:
```
ToolTipManager="*res://src/gui_common/tooltip/ToolTipManager.tscn"
```
So you can convert the quick load handler to be a scene that has the script attached.

Reply:
Ah, thank you! Didn't knew this

Reply:
I kept most of the logic in QuickLoad, because it is the right place for most of the logic.

Reply:
Ok, I've managed to create an AcceptDialog and open it from code, but I still don't see it. What am I missing?
I'm sorry, I've never used AutoLoad for stuff like that, thats new to me.

Reply:
All the autoloads are before the game content in the scene tree, so they are drawn underneath it. Check how overlay manager and the colourblind filter work to make the quick load handler be on top of the gameplay GUI but under the filter.

Reply:
![grafik](https://user-images.githubusercontent.com/30809803/97118540-c2307700-170a-11eb-89f6-36c98c7f0fba.png)
It seems like the style was not loaded at this point yet

Reply:
Each root Control needs the theme set. Theme is only a tscn file specific.
So you need to set the thrive theme as the theme on your root Control in the quick load scene.

Reply:
Ah, found it. Sorry for annoying you with those stupid questions